### ISSUE TITLE
nvbench::range drops the last element

### ISSUE THREAD

**Issue by PointKernel** [allisonvacanti/nvbench#3]
```cpp
void nvbench_test(nvbench::state& state)
{
  std::cout << state.get_float64("Occupancy") << "\n";
}

NVBENCH_BENCH(nvbench_test).add_float64_axis("Occupancy", nvbench::range(0.7, 0.9, 0.1));
```
Execution of the above code will work as expected and the output will be:
```
Run:  nvbench_test [Device=0 Occupancy=0.7]
0.7
Run:  nvbench_test [Device=0 Occupancy=0.8]
0.8
Run:  nvbench_test [Device=0 Occupancy=0.9]
0.9
```

If I change stride to `0.05`:
```cpp
NVBENCH_BENCH(nvbench_test).add_float64_axis("Occupancy", nvbench::range(0.7, 0.9, 0.05));
```
The output is shown below where the last element within the range (`0.9` ) is not included:
```
Run:  nvbench_test [Device=0 Occupancy=0.7]
0.7
Run:  nvbench_test [Device=0 Occupancy=0.75]
0.75
Run:  nvbench_test [Device=0 Occupancy=0.8]
0.8
Run:  nvbench_test [Device=0 Occupancy=0.85]
0.85
```

**Comment by allisonvacanti** [comment]
These are just floating point errors, you can work around them by making the max slightly higher

 I've hit these too and have been meaning to fix it. I'll have a patch in a few minutes to address this.

**Comment by allisonvacanti** [comment]
Fixed in 9f6404b.
